pkg/helpers/sa_syncer.go

```diff
@@ -0,0 +1,99 @@
+package helpers
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/openshift/library-go/pkg/operator/events"
+	"github.com/openshift/library-go/pkg/operator/resource/resourceapply"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	coreclientv1 "k8s.io/client-go/kubernetes/typed/core/v1"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/clientcmd"
+	clientcmdapi "k8s.io/client-go/tools/clientcmd/api"
+)
+
+// EnsureSAToken get the saToken of target sa and ensure it was rendered as expected.
+// A usage example combined with RenderToKubeconfigSecert would be like the following:
+// ...
+// saName := clusterManagerName+"-registration-controller-sa"
+// secretName := saName+"-kubeconfig"
+// err := helpers.EnsureSAToken(ctx, saName, clustermanagerNamespace, amdinKubeClient,
+// 	helpers.RenderToKubeconfigSecret(secretName, clustermanagerNamespace, AdminKubeconfig, kubeClient, recorder)) // kubeClient used to create secret; normally recorder is from controller object.
+// if err != nil {
+// 	return err
+// }
+// ...
+func EnsureSAToken(ctx context.Context, saName, saNamespace string, client kubernetes.Interface, renderSAToken func([]byte) error) error {
+	// get the service account
+	sa, err := client.CoreV1().ServiceAccounts(saNamespace).Get(ctx, saName, metav1.GetOptions{})
+	if err != nil {
+		return err
+	}
+	if len(sa.Secrets) == 0 {
+		return fmt.Errorf("token secret for %s not exist yet", saName)
+	}
+	for _, secret := range sa.Secrets {
+		if strings.Contains(secret.Name, "token") {
+			tokenSecretName := sa.Secrets[0].Name
```

Comment:
alway get the name of as.Secrets[0] here in loop.

Reply:
Solved.